client/my-sites/upgrades/domain-search/domain-search.jsx

```diff
@@ -139,7 +134,7 @@ var DomainSearch = React.createClass( {
 							selectedSite={ selectedSite }
```

Comment:
It seems `selectedSite` can indeed be `null` when we switch site (don't even need to clear the localStorage). I have updated propTypes to reflect that.
I think the work on site switching with an empty state is a much deeper issue and probably already breaks in many cases currently.
Here we have upgraded the component while keeping the same behavior which is already a step forward.